### PR TITLE
Pipe: Support "source.realtime.loose-range" = "path" in iotdb-source

### PR DESCRIPTION
**Pull request by SteveYurongSu**
As title.

### CODE REVIEW

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/extractor/dataregion/realtime/PipeRealtimeDataRegionExtractor.java [hunk 1]
```diff
@@ -142,10 +147,33 @@ public void validate(final PipeParameterValidator validator) throws Exception {
...
+                        EXTRACTOR_REALTIME_LOOSE_RANGE_DEFAULT_VALUE)
+                    .split(","))
+            .map(String::trim)
+            .map(String::toLowerCase)
+            .collect(Collectors.toSet());
+    // Avoid empty string
+    sloppyOptionSet.remove("");
+    sloppyTimeRange = sloppyOptionSet.contains(EXTRACTOR_REALTIME_LOOSE_RANGE_TIME_VALUE);
+    sloppyPattern = sloppyOptionSet.contains(EXTRACTOR_REALTIME_LOOSE_RANGE_PATH_VALUE);
+    if (!sloppyOptionSet.isEmpty()) {
```
Comment (VGalaxies) [inline iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/extractor/dataregion/realtime/PipeRealtimeDataRegionExtractor.java:172]: remove "time" and "path" before checking?

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/extractor/dataregion/realtime/PipeRealtimeDataRegionExtractor.java [hunk 2]
```diff
@@ -142,10 +147,33 @@ public void validate(final PipeParameterValidator validator) throws Exception {
                 EXTRACTOR_END_TIME_KEY,
                 realtimeDataExtractionEndTime));
       }
+    } catch (final PipeParameterNotValidException e) {
```
Comment (VGalaxies) [inline iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/extractor/dataregion/realtime/PipeRealtimeDataRegionExtractor.java:150]: Also, make changes here https://github.com/apache/iotdb/blob/master/iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/extractor/dataregion/historical/PipeHistoricalDataRegionTsFileExtractor.java#L186

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/extractor/dataregion/realtime/PipeRealtimeDataRegionExtractor.java (hunk 1 above)
Comment (SteveYurongSu) [reply iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/extractor/dataregion/realtime/PipeRealtimeDataRegionExtractor.java:172]: Good catch. Fixed.

**Review by VGalaxies — APPROVED** [review]

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/event/common/tablet/PipeInsertNodeTabletInsertionEvent.java [hunk 3]
```diff
@@ -205,29 +205,36 @@ public boolean isGeneratedByPipe() {
...
           return false;
         }
         // We assume that `timestamps` is ordered.
         return startTime <= timestamps[timestamps.length - 1] && timestamps[0] <= endTime;
-      } else if (insertNode instanceof InsertRowsNode) {
+      }
+
+      if (insertNode instanceof InsertRowsNode) {
         for (final InsertRowNode node : ((InsertRowsNode) insertNode).getInsertRowNodeList()) {
           final long timestamp = node.getTime();
```
[the remark this answers is not in the document]
Comment (SteveYurongSu) [reply iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/event/common/tablet/PipeInsertNodeTabletInsertionEvent.java]: Good idea. Applied.

**Review by Caideyipi — APPROVED** [review]

**Pull request merged**